src/Fixer/LanguageConstruct/FunctionToConstantFixer.php

```diff
@@ -0,0 +1,202 @@
+        'phpversion' => 'PHP_VERSION',
+        'php_sapi_name' => 'PHP_SAPI',
+        'pi' => 'M_PI',
+    );
+
+    /**
+     * @var array<string, string>
+     */
+    private $functionsFixMap;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        parent::configure($configuration);
+
+        $this->functionsFixMap = array();
+        foreach ($this->configuration['functions'] as $key) {
+            $this->functionsFixMap[$key] = self::$availableFunctions[$key];
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, Tokens $tokens)
+    {
+        for ($index = $tokens->count() - 4; $index > 0; --$index) {
+            $candidate = $this->getReplaceCandidate($tokens, $index);
+            if (null === $candidate) {
+                continue;
+            }
+
+            $this->fixFunctionCallToConstant(
+                $tokens,
+                $index,
+                $candidate[0], // brace open
+                $candidate[1], // brace close
+                $candidate[2]  // replacement
+            );
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Replace core functions calls returning constants with the constants.',
+            array(new CodeSample("<?php\necho phpversion();\necho pi();\necho php_sapi_name();")),
+            null,
+            'Risky when any of the configured functions to replace are overridden.'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // should run before NativeFunctionCasingFixer
+        // must run before NoExtraConsecutiveBlankLinesFixer, NoSinglelineWhitespaceBeforeSemicolonsFixer, NoTrailingWhitespaceFixer and NoWhitespaceInBlankLineFixer
+        // must run after NoSpacesAfterFunctionNameFixer and NoSpacesInsideParenthesisFixer
+
+        return 1;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(T_STRING);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isRisky()
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        $generator = new FixerOptionValidatorGenerator();
+        $functionNames = array_keys(self::$availableFunctions);
+        $functions = new FixerOptionBuilder('functions', 'List of function names to fix.');
+        $functions = $functions
+            ->setAllowedTypes(array('array'))
+            ->setAllowedValues(array(
+                $generator->allowedValueIsSubsetOf($functionNames),
+            ))
+            ->setDefault($functionNames)
+            ->getOption()
+        ;
+
+        return new FixerConfigurationResolverRootless('functions', array($functions));
```

Comment:
The `FixerConfigurationResolverRootless` is used to deprecate options that are not compatible with the new configuration resolver. As the option is new here, you should use `FixerConfigurationResolver` instead.